### PR TITLE
Add CrossToken ValidateToken test cases

### PR DESCRIPTION
**Pull request by ChenYan98**

### CODE REVIEW

File: test/Microsoft.IdentityModel.Tests/IdentityUtilities.cs [hunk 1]
```diff
@@ -93,6 +94,99 @@ public static JwtSecurityToken CreateJwtSecurityToken(string issuer, string audi
...
+            XmlWriter writer = XmlWriter.Create(sb);
+            tokenHandler.WriteToken(writer, securityToken);
+            writer.Flush();
+            writer.Close();
+            return sb.ToString();
+        }
+
+        public static string CreateSaml2SecurityToken(SecurityTokenDescriptor securityTokenDescriptor, SecurityTokenHandler tokenHandler)
+        {
+            SecurityToken securityToken = tokenHandler.CreateToken(securityTokenDescriptor);
```
Comment (brentschmaltz) [inline test/Microsoft.IdentityModel.Tests/IdentityUtilities.cs]: you could write this as:
var token = tokenHandler.CreateToken(securityTokenDescriptor)
return tokenHandler.WriteToken(token);

OR

return tokenHandler.WriteToken(tokenHandler.CreateToken(securityTokenDescriptor);

File: test/Microsoft.IdentityModel.Tests/IdentityUtilities.cs [hunk 2]
```diff
@@ -93,6 +94,99 @@ public static JwtSecurityToken CreateJwtSecurityToken(string issuer, string audi
             return new JwtSecurityToken(header, payload, header.Base64UrlEncode(), payload.Base64UrlEncode(), "" );
         }
 
+        public static string CreateSamlSecurityToken(SecurityTokenDescriptor securityTokenDescriptor, SecurityTokenHandler tokenHandler)
```
Comment (brentschmaltz) [inline test/Microsoft.IdentityModel.Tests/IdentityUtilities.cs]: normally we drop the 'security' prefix for variables: securityTokenDescriptor -> tokenDescriptor. This matches what we do for SecurityTokenHandler.

File: test/Microsoft.IdentityModel.Tests/IdentityUtilities.cs [hunk 3]
```diff
@@ -93,6 +94,99 @@ public static JwtSecurityToken CreateJwtSecurityToken(string issuer, string audi
...
+            XmlWriter writer = XmlWriter.Create(sb);
+            tokenHandler.WriteToken(writer, securityToken);
+            writer.Flush();
+            writer.Close();
+            return sb.ToString();
+        }
+
+        public static string CreateJwtSecurityToken(SecurityTokenDescriptor securityTokenDescriptor, SecurityTokenHandler tokenHandler)
+        {
+            SecurityToken securityToken = tokenHandler.CreateToken(securityTokenDescriptor);
```
Comment (brentschmaltz) [inline test/Microsoft.IdentityModel.Tests/IdentityUtilities.cs]: use 'var' when possible.

File: test/Microsoft.IdentityModel.Tests/IdentityUtilities.cs (hunk 2 above)
Comment (brentschmaltz) [inline test/Microsoft.IdentityModel.Tests/IdentityUtilities.cs]: Rename to CreateEncodedSaml.
The name distinguishes whether a token or string is returned.

File: test/Microsoft.IdentityModel.Tests/IdentityUtilities.cs [hunk 4]
```diff
@@ -93,6 +94,99 @@ public static JwtSecurityToken CreateJwtSecurityToken(string issuer, string audi
...
+            StringBuilder sb = new StringBuilder();
+            XmlWriter writer = XmlWriter.Create(sb);
+            tokenHandler.WriteToken(writer, securityToken);
+            writer.Flush();
+            writer.Close();
+            return sb.ToString();
+        }
+
+        public static string CreateSaml2SecurityToken(SecurityTokenDescriptor securityTokenDescriptor, SecurityTokenHandler tokenHandler)
```
Comment (brentschmaltz) [inline test/Microsoft.IdentityModel.Tests/IdentityUtilities.cs]: Rename to CreateEncodedSaml2.
The name distinguishes whether a token or string is returned.

File: test/Microsoft.IdentityModel.Tests/IdentityUtilities.cs [hunk 5]
```diff
@@ -93,6 +94,99 @@ public static JwtSecurityToken CreateJwtSecurityToken(string issuer, string audi
...
+            return sb.ToString();
+        }
+
+        public static string CreateJwtSecurityToken(SecurityTokenDescriptor securityTokenDescriptor, SecurityTokenHandler tokenHandler)
+        {
+            SecurityToken securityToken = tokenHandler.CreateToken(securityTokenDescriptor);
+            return tokenHandler.WriteToken(securityToken);
+        }
+
+        public static SecurityTokenDescriptor DefaultX509SecurityTokenDescriptor
```
Comment (brentschmaltz) [inline test/Microsoft.IdentityModel.Tests/IdentityUtilities.cs]: A lot of these 'Default' methods are duplicates of what is in Default.cs. 
Remove them and If something is missing Default.cs, add the property there.

File: test/Microsoft.IdentityModel.Tokens.Tests/CrossTokenTests.cs [hunk 6]
```diff
@@ -0,0 +1,116 @@
...
+    /// results in the same exceptions, claims etc.
+    /// </summary>
+    public class CrossTokenTests
+    {
+#pragma warning disable CS3016 // Arrays as attribute arguments is not CLS-compliant
+        [Theory, MemberData("CrossTokenValidateTokenTheoryData")]
+#pragma warning restore CS3016 // Arrays as attribute arguments is not CLS-compliant
+        public void CrossTokenValidateToken(CrossTokenTheoryData theoryData)
+        {
+            TestUtilities.WriteHeader($"{this}.CrossTokenValidateToken", theoryData);
```
Comment (brentschmaltz) [inline test/Microsoft.IdentityModel.Tokens.Tests/CrossTokenTests.cs]: TestUtilities.WriteHeader returns a CompareContext. 
Use: context = TestUtilities.WriteHeader(...)

File: test/Microsoft.IdentityModel.Tests/IdentityUtilities.cs (hunk 2 above)
Comment (ChenYan98) [reply test/Microsoft.IdentityModel.Tests/IdentityUtilities.cs]: Fixed

Comment (ChenYan98) [reply test/Microsoft.IdentityModel.Tests/IdentityUtilities.cs]: Fixed

File: test/Microsoft.IdentityModel.Tests/IdentityUtilities.cs (hunk 4 above)
Comment (ChenYan98) [reply test/Microsoft.IdentityModel.Tests/IdentityUtilities.cs]: Fixed

File: test/Microsoft.IdentityModel.Tests/IdentityUtilities.cs (hunk 1 above)
Comment (ChenYan98) [reply test/Microsoft.IdentityModel.Tests/IdentityUtilities.cs]: Fixed

File: test/Microsoft.IdentityModel.Tests/IdentityUtilities.cs (hunk 3 above)
Comment (ChenYan98) [reply test/Microsoft.IdentityModel.Tests/IdentityUtilities.cs]: Remove this variables and merge it to next line.

File: test/Microsoft.IdentityModel.Tests/IdentityUtilities.cs (hunk 5 above)
Comment (ChenYan98) [reply test/Microsoft.IdentityModel.Tests/IdentityUtilities.cs]: Fixed

File: test/Microsoft.IdentityModel.Tokens.Tests/CrossTokenTests.cs (hunk 6 above)
Comment (ChenYan98) [reply test/Microsoft.IdentityModel.Tokens.Tests/CrossTokenTests.cs]: Fixed

**Comment by brentschmaltz** [comment]
:shipit:

**Pull request merged**